### PR TITLE
media: Allow extra MIME parameters in decodingInfo

### PR DESCRIPTION
**Pull request by borongc**
Cobalt web applications frequently append custom MIME parameters to
strings, such as enableflushduringseek. The default implementation
is restrictive, failing validation if multiple parameters are
present or if the codecs parameter is not in the primary position.

This ensures MediaCapabilities correctly handles extra parameters,
preventing valid media configurations from being rejected due to
platform-specific metadata.

Issue: 519196649
Issue: 537784933

### CODE REVIEW

File: third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc [hunk 1]
```diff
@@ -287,13 +287,27 @@ bool IsValidMimeType(const String& content_type,
...
+  // Check across all parameters rather than assuming 'codecs' is the first parameter.
+  if (parameters.ParameterCount() == 0)
+    return true;
+
+  for (const auto& param : parameters) {
+    if (EqualIgnoringASCIICase(param.name, kCodecsMimeTypeParam))
+      return true;
+  }
+  return true;
+#else  // BUILDFLAG(IS_COBALT)
```
[the remark this answers is not in the document]
Comment (borongc) [reply third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc:302]: Done.

**Review by jasonzhangxx — APPROVED** [review]

File: third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc [hunk 2]
```diff
@@ -287,13 +287,27 @@ bool IsValidMimeType(const String& content_type,
...
+  // in any order (e.g. enableflushduringseek=true, enableresetaudiodecoder=true).
+  // Check across all parameters rather than assuming 'codecs' is the first parameter.
+  if (parameters.ParameterCount() == 0)
+    return true;
+
+  for (const auto& param : parameters) {
+    if (EqualIgnoringASCIICase(param.name, kCodecsMimeTypeParam))
+      return true;
+  }
+  return false;
```
Comment (hlwarriner) [inline third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc:301]: Just confirming that this logic is intended? It looks a bit surprising to me.

* If there are no parameters, consider it valid
* If there are parameters, and any is a mime type param, consider it valid
* If there are parameters, and none is a mime type param, consider it invalid

**Review by hlwarriner — APPROVED** [review]

File: third_party/blink/renderer/modules/media_capabilities/media_capabilities_test.cc [hunk 3]
```diff
@@ -654,6 +654,36 @@ TEST(MediaCapabilitiesTests, BasicAudio) {
   EXPECT_TRUE(info->powerEfficient());
 }
 
+#if BUILDFLAG(IS_COBALT)
+TEST(MediaCapabilitiesTests, CobaltExtraMimeParameters) {
```
Comment (hlwarriner) [inline third_party/blink/renderer/modules/media_capabilities/media_capabilities_test.cc:658]: Thanks for adding the unit test. If it's doable, consider also adding a test case for when there are parameters, but none are mime type params, and so `IsValidMimeType()` returns false.

**Review by hlwarriner — APPROVED** [review]

File: third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc (hunk 2 above)
Comment (borongc) [reply third_party/blink/renderer/modules/media_capabilities/media_capabilities.cc:301]: Yes, this is intended as what Chromium does, except that for Cobalt, we check each MIME type, not just the first one.

File: third_party/blink/renderer/modules/media_capabilities/media_capabilities_test.cc (hunk 3 above)
Comment (borongc) [reply third_party/blink/renderer/modules/media_capabilities/media_capabilities_test.cc:658]: Done.
